Dockerfile

```diff
@@ -0,0 +1,37 @@
+
+FROM ruby:3.1.2
+
+# Install Node.js
+RUN curl -sL https://deb.nodesource.com/setup_20.x | bash -
+RUN apt-get install -y nodejs
+
+# Install yarn version 3.2.1
+RUN node -v && npm i -g yarn && yarn set version 3.2.1
+# Set the working directory in the container
+WORKDIR /app
+
+# Install foreman to leverage Procfile
+RUN gem install foreman
+
+# Copy node package maps
+COPY .yarnrc.yml package.json yarn.lock /app/
+COPY .yarn/ /app/.yarn/
+
+COPY Gemfile* /app
+# Install Gemfile
+RUN bundle install
+
+# Install node modules
+RUN yarn install
+
+# Copy the current directory contents into the container at /app
+COPY . /app
+
+# Expose the rails app to outside the container
+EXPOSE 3000
+
+# Expose hmr server to outside the container
+EXPOSE 3035 
+
+# Run the app when the container launches
+CMD ["foreman", "start", "-f",  "Procfile.dev"]
```

Comment:
This could be configured differently, but it is likely a good idea to try to run both bare metal and containers through the same commands to avoid the need to update in multiple places when change comes.